.github/workflows/win.yml

```diff
@@ -0,0 +1,134 @@
+name: Windows CI
+on:
+  push:
+    paths-ignore:
+      - '*.md'
+      - .circleci/
+      - bench/
+      - docs/
+      - run_route_scripts/
+      - test/
+      - test_requests/
+    branches:
+      - master
+  pull_request:
+    paths-ignore:
+      - '*.md'
+      - .circleci/
+      - bench/
+      - docs/
+      - run_route_scripts/
+      - test/
+      - test_requests/
+    branches:
+      - master
+  workflow_dispatch:
+    inputs:
+      debug_enabled:
+        type: boolean
+        description: 'Run the build with tmate debugging enabled (https://github.com/marketplace/actions/debugging-with-tmate)'
+        required: false
+        default: false
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  BUILD_TYPE: Release
+  MSVC_VERSION: '2022'
+  VCPKG_VERSION: '2024.01.12'
+  VCPKG_INSTALL_OPTIONS: --x-abi-tools-use-exact-versions
+  VCPKG_DISABLE_COMPILER_TRACKING: ON
```

Comment:
these 2 are supposedly making sure that we have less cache misses if the vcpkg toolchain was updated, e.g newer cmake